File: tremendoclib/src/main/java/com/e/tremendocSDK/View/UI/Activity/Signup.java
package com.e.tremendocSDK.View.UI.Activity;

//import android.support.v7.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.e.tremendocSDK.Api.API;
import com.e.tremendocSDK.Api.StringCall;
import com.e.tremendocSDK.Api.URLS;
import com.e.tremendocSDK.R;
import com.e.tremendocSDK.Service.Model.SDK_User;
import com.e.tremendocSDK.View.Callback.FragmentChanger;
import com.e.tremendocSDK.View.Callback.ModelSaver;
import com.e.tremendocSDK.View.UI.Fragment.FragmentTitled;
import com.e.tremendocSDK.View.UI.Fragment.Signup.Stepone;
import com.e.tremendocSDK.View.UI.Fragment.Signup.Steptwo;
import com.e.tremendocSDK.View.UI.UUitil.Checker;
import com.e.tremendocSDK.View.UI.UUitil.DeviceName;
import com.e.tremendocSDK.View.UI.UUitil.IO;
import com.e.tremendocSDK.View.UI.UUitil.ToastUtili;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class Signup extends AppCompatActivity implements FragmentChanger, ModelSaver<SDK_User> {
    public static final String STEP_ONE = "step_one";
    public static final String STEP_TWO = "step_two";
    private SDK_User sdk_user = new SDK_User();
    private StringCall call;
    private ProgressDialog dialog;
    private Checker checker;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_auth);
        changeView(STEP_ONE);
        dialog = new ProgressDialog(this);
        checker=new Checker(this);
    }

    @Override
    public void ChangeFragment(FragmentTitled fragment) {
        changeView(fragment);
    }

    private void changeView(Fragment fragment) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.frame, fragment);
        fragmentTransaction.commit();
    }

    private void changeView(String fragmentName) {
        Fragment fragment;
        setTitle(fragmentName);
        switch (fragmentName) {
            case STEP_ONE:
                fragment = Stepone.newInstance();
                break;
            case STEP_TWO:
                fragment = Steptwo.newInstance();
                break;
            default:
                fragment = Stepone.newInstance();
        }
        changeView(fragment);

    }

    @Override
    public void onSave(SDK_User sdk_user) {
        this.sdk_user = sdk_user;
        if (checker.isOnline()) {
            Signup();
        }

    }


    private void Signup() {

        dialog = ProgressDialog.show(this, "Sign Up", "Please Wait....", false, false);

        call = new StringCall(this);
        Map<String, String> loadData = sdk_user.toMap();
        loadData.put("operatingSystem", "ANDROID");
        loadData.put("uuid", IO.getData(this, API.MY_UUID));
        loadData.put("brand", DeviceName.getDeviceName());
        loadData.put("sdkType","chat");
        loadData.put("provider","1");

        call.post(URLS.SDK_CREATE_USER, loadData, response -> {
            try {
                dialog.dismiss();
                JSONObject obj = new JSONObject(response);
                if (obj.has("code") && obj.getInt("code") == 0) {


                    Intent intent = new Intent(this, Finddoctor.class);
                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
                    startActivity(intent);
                    finish();

                } else {
                    dialog.dismiss();
                    String dec = obj.getString("description");
                    Toast.makeText(this, dec, Toast.LENGTH_LONG).show();
                }

            } catch (JSONException e) {
                ToastUtili.showModal(this, e.getMessage());

            }


        }, error -> {
            dialog.dismiss();

            if (error.networkResponse == null) {
                ToastUtili.showModal(this, "Please check your internet connection");
            }
        });
    }


}

File: tremendoclib/src/main/java/com/e/tremendocSDK/View/UI/Activity/Signup.kt
package com.e.tremendocSDK.View.UI.Activity

//import android.support.v7.app.AppCompatActivity;

import android.app.ProgressDialog
import android.content.Intent
import android.os.Bundle
import android.widget.Toast

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction

import com.e.tremendocSDK.Api.API
import com.e.tremendocSDK.Api.StringCall
import com.e.tremendocSDK.Api.URLS
import com.e.tremendocSDK.R
import com.e.tremendocSDK.Service.Model.SDK_User
import com.e.tremendocSDK.View.Callback.FragmentChanger
import com.e.tremendocSDK.View.Callback.ModelSaver
import com.e.tremendocSDK.View.UI.Fragment.FragmentTitled
import com.e.tremendocSDK.View.UI.Fragment.Signup.Stepone
import com.e.tremendocSDK.View.UI.Fragment.Signup.Steptwo
import com.e.tremendocSDK.View.UI.UUitil.Checker
import com.e.tremendocSDK.View.UI.UUitil.DeviceName
import com.e.tremendocSDK.View.UI.UUitil.IO
import com.e.tremendocSDK.View.UI.UUitil.ToastUtili

import org.json.JSONException
import org.json.JSONObject

import java.util.HashMap

class Signup : AppCompatActivity(), FragmentChanger, ModelSaver<SDK_User> {
    private var sdk_user = SDK_User()
    private var call: StringCall? = null
    private var dialog: ProgressDialog? = null
    private var checker: Checker? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_auth)
        changeView(STEP_ONE)
        dialog = ProgressDialog(this)
        checker = Checker(this)
    }

    override fun ChangeFragment(fragment: FragmentTitled) {
        changeView(fragment)
    }

    private fun changeView(fragment: Fragment) {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.signupFrame, fragment)
        fragmentTransaction.commit()
    }

    private fun changeView(fragmentName: String) {
        val fragment: Fragment
        title = fragmentName
        when (fragmentName) {
            STEP_ONE -> fragment = Stepone.newInstance()
            STEP_TWO -> fragment = Steptwo.newInstance()
            else -> fragment = Stepone.newInstance()
        }
        changeView(fragment)

    }

    override fun onSave(sdk_user: SDK_User) {
        this.sdk_user = sdk_user
        if (checker!!.isOnline) {
            Signup()
        }

    }


    private fun Signup() {

        dialog = ProgressDialog.show(this, "Sign Up", "Please Wait....", false, false)

        call = StringCall(this)
        val loadData = sdk_user.toMap()
        loadData["operatingSystem"] = "ANDROID"
        loadData["uuid"] = IO.getData(this, API.MY_UUID)
        loadData["brand"] = DeviceName.getDeviceName()
        loadData["sdkType"] = "chat"
        loadData["provider"] = "1"

        call!!.post(URLS.SDK_CREATE_USER, loadData, { response ->
            try {
                dialog!!.dismiss()
                val obj = JSONObject(response)
                if (obj.has("code") && obj.getInt("code") == 0) {


                    val intent = Intent(this, Finddoctor::class.java)
                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP)
                    startActivity(intent)
                    finish()

                } else {
                    dialog!!.dismiss()
                    val dec = obj.getString("description")
                    Toast.makeText(this, dec, Toast.LENGTH_LONG).show()
                }

            } catch (e: JSONException) {
                ToastUtili.showModal(this, e.message)

            }


        }, { error ->
            dialog!!.dismiss()

            if (error.networkResponse == null) {
                ToastUtili.showModal(this, "Please check your internet connection")
            }
        })
    }

    companion object {

        val STEP_ONE = "step_one"
        val STEP_TWO = "step_two"
    }


}
